### PR TITLE
Add ability to for setting persistence through CR (during installation)

### PR DESCRIPTION
**Pull request by waveywaves**
- Add spec option for Persistence

- Update reconciler to handle persistence

- Add CustomResource Artifacts for ephemeral and persistent jenkins

- Made Persistence.enabled a required field in CRD

- Add ability to set the size of PV

### CODE REVIEW

File: pkg/controller/jenkins/reconciler.go [hunk 1]
```diff
@@ -211,6 +214,13 @@ func (r *JenkinsReconciler) checkResourceIfExists(resource RuntimeResource) erro
 	return err
 }
 
+func (r *JenkinsReconciler) isPersistent() bool {
+	if r.ControlledRescources.JenkinsInstance.Spec.Persistence.Enabled {
+		return true
+	}
+	return false
+}
+
```
Comment (akram) [inline pkg/controller/jenkins/reconciler.go]: You can simplify this into `return  r.ControlledRescources.JenkinsInstance.Spec.Persistence.Enabled`

**Comment by akram** [comment]
other this things looks good to me

File: pkg/controller/jenkins/reconciler.go (hunk 1 above)
Comment (waveywaves) [reply pkg/controller/jenkins/reconciler.go]: updated

**Pull request merged**